### ISSUE TITLE
RoomListService and SyncService states mismatching

### ISSUE THREAD

**Issue by ganfra** [matrix-org/matrix-rust-sdk#2365]
When encryption sync is enabled like this : 
```
 val syncService = client.syncService()
            .withEncryptionSync(false, null)
            .finish()
```

and we enter airplane mode, I'd expect both `RoomListService.State` and `SyncService.State` to be in `Error`.
But it's not the case, only the `SyncService.State` is updated.
If the encryption sync is not enabled, the state is updated for both services.

Some logs when I enter airplane : 

```
17:28:56.905 org.ma...st.sdk  D  sync_stream{conn_id="encryption" with_e2ee=true}:sync_once{pos="2"}:send{homeserver="https://slidingsync.lab.matrix.org/" server_versions=[V1_0, V1_1, V1_2, V1_3, V1_4, V1_5, V1_6] config=RequestConfig
                                 { timeout: 60s
                                 } request_id="REQ-9" request_size="71
                                 B" request_body=b"{\"conn_id\":\"encryption\",\"extensions\":{\"to_device\":{\"since\":\"5758901\"}}}" path="/_matrix/client/unstable/org.matrix.msc3575/sync?pos=2&timeout=30000"}: matrix_sdk::http_client: crates/matrix-sdk/src/http_client/mod.rs:208: Error
                                 while sending
                                 request: Reqwest(reqwest::Error
                                 { kind: Request, url: Url
                                 { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("slidingsync.lab.matrix.org")), port: None, path: "/_matrix/client/unstable/org.matrix.msc3575/sync", query: Some("pos=2&timeout=30000"), fragment: None
                                 }, source: hyper::Error(Io, Kind(ConnectionAborted)) })
17:28:56.906                  E  matrix_sdk_ui::sync_service: crates/matrix-sdk-ui/src/sync_service/mod.rs:133: Encryption
                                 sync returned an
                                 error: Something wrong happened
                                 in sliding sync: error sending
                                 request for url
                                 (https://slidingsync.lab.matrix.org/_matrix/client/unstable/org.matrix.msc3575/sync?pos=2&timeout=30000): connection
                                 error: connection aborted

```

cc @Hywan @bnjbvr

**Comment by bnjbvr** [comment]
We should probably hide the entire encryption sync from the user, and only expose the room list state. In the meanwhile, it's strange that the room list doesn't transition to the terminated state (because the encryption sync failed, this should stop the room list service in the sync service):

**Comment by bnjbvr** [comment]
So there was one bug that's going to be fixed by #2362 (where the sync service would transition from "Error" -> "Terminated" incorrectly). Now, if I read the description correctly, it is suggested that the "child" sync (RoomListService) state should transition to the Error state whenever the "parent" service (SyncService) runs into an error. I don't think that's the good way to go here, or that it should matter: one should observe the SyncService's state to decide whether to restart it or not, not the child's state. Users of the `SyncService` should:

- restart immediately after hitting the `Terminated`/`Idle` state; that means the previous sliding syncs naturally stopped OR the sync service was `pause()`d.
- restart after a small wait (incremented with an exponential backoff policy, likely) after hitting the `Error` state.

As such, closing this issue; we can open new ones if something else is broken or wrong.
